dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/OpenTelemetryProducer.java

```diff
@@ -89,6 +99,25 @@ public OpenTelemetry getOpenTelemetry() {
 
     }
 
+    //Uses application name if the user has not given configured service.name resource attribute
+    private String getServiceName() {
+        String appName = config.getOptionalValue(SERVICE_NAME_PROPERTY, String.class).orElse(null);
```

Comment:
At this point, it probably makes more sense to use the `ConfigProperties` object to get the service name, rather than going back to MP Config.

We already put everything from MP Config into the OTel's `ConfigProperties` and we'd expect everything else at this point to use `ConfigProperties`.

Reply:
This part hasn't been changed.

Reply:
It's in the last commit here: 

https://github.com/OpenLiberty/open-liberty/pull/23892/commits/35a4d8eedd010926f93f4894c9aa786b1de9fbad

Reply:
Ah, thank you, I see it now.